src/main/java/leekscript/compiler/WordCompiler.java

```diff
@@ -922,50 +903,46 @@ private void variableDeclaration(LeekType type) throws LeekCompilerException {
 		}
 		LeekVariableDeclarationInstruction variable = new LeekVariableDeclarationInstruction(this, word, getCurrentFunction(), type);
 		// On regarde si une valeur est assignée
-		if (mCompiler.haveWords() && mCompiler.token().getWord().equals("=")) {
-			mCompiler.skipToken();
+		if (mTokens.hasMoreTokens() && mTokens.get().getWord().equals("=")) {
+			mTokens.skip();
 
 			// Arrow function?
-			int p = mCompiler.findNextClosingParenthesis();
-			int a = mCompiler.findNextArrow();
-			int c = mCompiler.findNextColon();
-			boolean parenthesis = false;
-			boolean isArrowFunction = parenthesis || (a != -1 && (a < p || p == -1) && (a < c || c == -1));
```

Comment:
This is the only code that changed in this file, the rest is purely renaming

Reply:
`findNextColon` has been removed because the old (and new) lexical parsers don't parse ":" as T_COLON, but T_OPERATOR. So `findNextColon` would always return -1. I simplified the boolean expressions.